crates/wasmtime/src/engine.rs

```diff
@@ -655,6 +661,11 @@ impl Engine {
         &self.inner.signatures
     }
 
+    #[cfg(feature = "runtime")]
+    pub(crate) fn custom_code_memory(&self) -> Option<Arc<dyn CustomCodeMemory>> {
```

Comment:
Would it be possible to return `Option<&dyn ...>` here to avoid forcing the clone on callers? Or perhaps `Option<&Arc<dyn ...>>` if callers should optionally have the ability to create an owned copy